p2p/dial.go

```diff
@@ -0,0 +1,130 @@
+package p2p
+
+import (
+	"crypto/rand"
+	"fmt"
+	"net"
+
+	"github.com/ethereum/go-ethereum/logger"
+	"github.com/ethereum/go-ethereum/logger/glog"
+	"github.com/ethereum/go-ethereum/p2p/discover"
+)
+
+// dialstate creates dials and discovery lookups.
+type dialstate struct {
+	lookupBuf     []*discover.Node // current discovery lookup results
+	lookupRunning bool
+
+	static      map[discover.NodeID]*discover.Node
+	dialing     map[discover.NodeID]connFlag
+	maxDynDials int
+}
+
+func newDialState(static []*discover.Node, maxdyn int) *dialstate {
+	s := &dialstate{
+		maxDynDials: maxdyn,
+		static:      make(map[discover.NodeID]*discover.Node),
+		dialing:     make(map[discover.NodeID]connFlag),
+	}
+	for _, n := range static {
+		s.static[n.ID] = n
+	}
+	return s
+}
+
+func (s *dialstate) addStatic(n *discover.Node) {
+	s.static[n.ID] = n
+}
+
+func (s *dialstate) newTasks(peers map[discover.NodeID]*Peer) []task {
```

Comment:
I think we should consider a different name for this method. newSomething is usually used to construct an object, but here we are actually devising a strategy for the dialer. It's arguable that the strategy is made up of tasks, but if I just glance at the code it's not what I would expect.

The least invasive mod would probably be `makeTasks`, and it's already imho a bit more reflective that it's not a simple constructor. Maybe `suggestTasks` would be even better. (If at least the dialer has a choice to discard these (I'm guessing it does, but I've yet to look through that part in detail).